### PR TITLE
Cntools 7.2.3

### PR DESCRIPTION
**Pull request by Scitz0**
Redesign of backup and restore.
- Deletion of private keys moved from backup to new section under `Advanced`
- Backup now only contain content of priv folder (files & scripts folders dropped)
- Restore operation now restore directly to priv folder instead of a random user selected folder to make restore easier and better. Before restore, a new full backup of priv folder is made and stored encrypted in priv/archive
- Various improvements to backup & restore

### CODE REVIEW

**Comment by TrevorBenson** [comment]
A few comments after being too busy for a couple of weeks:

> Redesign of backup and restore.
> 
> * Deletion of private keys moved from backup to new section under `Advanced`
> * Backup now only contain content of priv folder (files & scripts folders dropped)

Very happy to see this, never opened a request but had it on my mind while testing manual restore processes.

> * Restore operation now restore directly to priv folder instead of a random user selected folder to make restore easier and better. Before restore, a new full backup of priv folder is made and stored encrypted in priv/archive

I love the level of consideration put into this change, it really shows consideration around an SPO making a dire mistake.

That said, and while I know you're thorough in your work, as we are talking about changes to backups and to restore processes I'm going to build a test system and pull the branch to test the backup and restore process before stamping the "LGTM".

I won't get to this until later tonight, but obviously won't object to others approving your changes and merging if they get around to testing it first.

> * Various improvements to backup & restore

Thanks @Scitz0 , for continually improving the tools,

**Comment by Scitz0** [comment]
> That said, and while I know you're thorough in your work, as we are talking about changes to backups and to restore processes I'm going to build a test system and pull the branch to test the backup and restore process before stamping the "LGTM".
> 
> I won't get to this until later tonight, but obviously won't object to others approving your changes and merging if they get around to testing it first.

🙏 It was actually Priyanks idea to make the change and how. I liked it and implemented it 😄

Your mindset is correct, should not approve stuff without proper review and testing, even if it's a merge to alpha. I have done some testing while implementing it but far from every use case and combination of options. Easy to make logical mistakes and/or miss to handle edge cases.

**Review by TrevorBenson — APPROVED** [review]
Tested both defaults on bare-metal and containers w/ non default values (CHATTR=false etc.) working exactly as described.

LGTM!

File: scripts/cnode-helper-scripts/cntools.sh [hunk 1]
```diff
@@ -4461,6 +4503,66 @@ EOF
...
+    println "DEBUG" "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+    echo
+    
+    println "DEBUG" "The following files will be removed"
+    println "DEBUG" "Wallet ${FG_LGRAY}${WALLET_PAY_SK_FILENAME}${NC} / ${FG_LGRAY}${WALLET_STAKE_SK_FILENAME}${NC}"
+    println "DEBUG" "Pool   ${FG_LGRAY}${POOL_COLDKEY_SK_FILENAME}${NC}"
+    [[ -d "${ASSET_FOLDER}" ]] && println "DEBUG" "Asset  ${FG_LGRAY}${ASSET_POLICY_SK_FILENAME}${NC}"
+    echo
+    
+    println "DEBUG" "${FG_RED}Do you acknowledge that you have already taken a full backup, and are OK to simply delete the private keys? There is no going back !!!${NC}"
```
Comment (rdlrt) [inline scripts/cnode-helper-scripts/cntools.sh:4520]: Thanks!

**Review by rdlrt — APPROVED** [review]
LGTM!

File: scripts/cnode-helper-scripts/cntools.library [hunk 2]
```diff
@@ -11,7 +11,7 @@ CNTOOLS_MAJOR_VERSION=7
 # Changes that can be applied without breaking existing functionality that can be applied from within CNTools
 CNTOOLS_MINOR_VERSION=2
 # Backwards compatible bug fixes. No additional functionality or major changes and can be applied from within CNTools
-CNTOOLS_PATCH_VERSION=2
+CNTOOLS_PATCH_VERSION=3
```
[the remark this answers is not in the document]
Comment (Scitz0) [reply scripts/cnode-helper-scripts/cntools.library]: Yip, sounds good. The MA addition together with the backup change warrants a major version bump.

**Pull request merged**